PyCover.py

```diff
@@ -11,126 +11,133 @@
-                         stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    threading.Thread(target=missing_lines_callback, args=(self.view, p)).start()
+
+    """Highlight uncovered lines in the current file
+    based on a previous coverage run."""
+
+    def is_visible(self):
+        return self.is_enabled()
+
+    def is_enabled(self):
+        return 'source.python' in self.view.scope_name(0)
+
+    def run(self, edit):
+        fname = self.view.file_name()
+        if not self.is_enabled() or not fname:
+            return
+
+        local_settings = self.view.settings()
+        if local_settings.get('showing', False):
+            self.view.erase_regions('PyCover')
+            local_settings.set('showing', False)
+            return  # Toggle off
+
+        cov_file = find(fname, '.coverage')
+        if not cov_file:
+            status_report('Could not find .coverage file for %s' % fname, wrap=True)
+            return
+        cov_config = find(fname, '.coveragerc') or ''
+
+        # run missing_lines.py with the correct paths
+        python = SETTINGS.get('python', '')
+        if not python:
+            python = which('python')
+        ml_file = os.path.join(sublime.packages_path(), 'PyCover', 'scripts',
+                               'missing_lines.py')
+        p = subprocess.Popen([python, ml_file, cov_file, cov_config, fname],
+                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        threading.Thread(target=missing_lines_callback, args=(self.view, p)).start()
 
 
 def missing_lines_callback(view, proc, poll_sleep=0.1, poll_timeout=10):
-  progress_status = lambda: sublime.status_message('Finding missing lines...')
-  sublime.set_timeout(progress_status, 0)
-  # poll for results
-  tic = time.time()
-  while proc.poll() is None:
-    if time.time() - tic > poll_timeout:
-      msg = 'missing_lines.py timed out after %f s' % (time.time() - tic)
-      status_report(msg, wrap=True)
-      proc.kill()
-      return
-    time.sleep(poll_sleep)
+    progress_status = lambda: sublime.status_message('Finding missing lines...')
     sublime.set_timeout(progress_status, 0)
+    # poll for results
+    tic = time.time()
+    while proc.poll() is None:
+        if time.time() - tic > poll_timeout:
+            msg = 'missing_lines.py timed out after %f s' % (time.time() - tic)
+            status_report(msg, wrap=True)
+            proc.kill()
+            return
+        time.sleep(poll_sleep)
+        sublime.set_timeout(progress_status, 0)
 
-  stdout, stderr = proc.communicate()
-  if proc.returncode != 0:
-    status_report(stderr.decode('UTF-8'), wrap=True)
-    return
+    stdout, stderr = proc.communicate()
+    if proc.returncode != 0:
+        status_report(stderr.decode('UTF-8'), wrap=True)
+        return
 
-  # read stdout to parse missing lines
-  missing_lines = map(int, stdout.decode('UTF-8').splitlines())
+    if b'omitted file' in stdout:
```

Comment:
A test for equality here (or maybe `startswith`) would be better here. The `in` operator will scan over the whole string, which could be long.